### PR TITLE
Validate file name in codeDir for PAI platform

### PR DESCRIPTION
**Pull request by SparkSnail**
https://github.com/microsoft/nni/issues/1164

### CODE REVIEW

File: src/nni_manager/training_service/common/util.ts [hunk 1]
```diff
@@ -40,9 +40,10 @@ import { file } from "../../node_modules/@types/tmp";
  */
 export async function validateCodeDir(codeDir: string) : Promise<number> {
     let fileCount: number | undefined;
-
+    let fileNameValid: boolean = true;
     try {
         fileCount = await countFilesRecursively(codeDir);
+        fileNameValid = await validateFileNameRecursively(codeDir);
     } catch(error) {
         throw new Error(`Call count file error: ${error}`);
```
Comment (QuanluZhang) [inline src/nni_manager/training_service/common/util.ts]: this error message is not proper any more. Suggest to do countFile and validateFile one by one separately.

File: src/nni_manager/common/utils.ts [hunk 2]
```diff
@@ -374,6 +374,40 @@ function countFilesRecursively(directory: string, timeoutMilliSeconds?: number):
...
+    }
+
+    const fileNameArray: string[] = fs.readdirSync(directory);
+    let result = true;
+    for(var fileName of fileNameArray){
+        const fullFilePath: string = path.join(directory, fileName);
+        try {
+            if (fs.lstatSync(fullFilePath).isFile()) {
+                result = validateFileName(fileName);
+            } else {
```
Comment (QuanluZhang) [inline src/nni_manager/common/utils.ts]: also need to validate directory's name?

File: src/nni_manager/training_service/common/util.ts (hunk 1 above)
Comment (SparkSnail) [reply src/nni_manager/training_service/common/util.ts]: fixed, validate filename and filecount separately.

File: src/nni_manager/common/utils.ts (hunk 2 above)
Comment (SparkSnail) [reply src/nni_manager/common/utils.ts]: fixed, validate directory name.

**Pull request merged**